api_app/analyzers_manager/observable_analyzers/knockanalyzer.py

```diff
@@ -0,0 +1,101 @@
+import glob
+import json
+import logging
+import os
+
+from knock.knockpy import KNOCKPY
+
+from api_app.analyzers_manager import classes
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class KnockAnalyzer(classes.ObservableAnalyzer):
+    """
+    This analyzer is a wrapper for the knockpy project.
+    """
+
+    def update(self) -> bool:
+        pass
+
+    observable_name: str
+    dns: str = None
+    useragent: str = None
+    timeout: int = None
+    threads: int = None
+    recon: bool = True
+    bruteforce: bool = True
+
+    def run(self):
+        logger.info(f"Running KnockAnalyzer for {self.observable_name}")
+        results = KNOCKPY(
+            domain=self.observable_name,
+            dns=self.dns,
+            useragent=self.useragent,
+            timeout=self.timeout,
+            threads=self.threads,
+            recon=self.recon,
+            bruteforce=self.bruteforce,
+        )
+        files = glob.glob("domain.com*.json")
+        for file in files:
+            logger.info(f"Removing {file}")
+            os.remove(file)
```

Comment:
There is a possibility to save the results only in memory? 
I hate to have to save data in files, read them again, and delete.
It is a really bad pattern